src/core/time_arrays.jl

```diff
@@ -272,64 +459,220 @@ split_indices(idxs, ti) = idxs[1:ti - 1], idxs[ti], idxs[ti + 1:end]
 	idxs1, ts, idxs2 = split_indices(idxs, ti)
-	t = ts.t + findfirst(isequal(T_TIMES[1]), D_TIMES) - 1
-	return arr.data[idxs1..., t, idxs2...]
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		t = ts.t + findfirst(isequal(T_TIMES[1]), D_TIMES) - 1
+		return arr.data[idxs1..., t, idxs2...]
+	end
+end
+
+function Base.getindex(arr::TimestepArray{FixedTimestep{FIRST, STEP}, T_data, N, ti}, idxs::Union{TimestepValue{T_time}, AnyIndex}...) where {T_data, N, ti, FIRST, STEP, T_time}
+	_single_index_check(arr.data, idxs)
+	idxs1, ts, idxs2 = split_indices(idxs, ti)
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		LAST = FIRST + ((size(arr.data, ti) - 1) * STEP)
+		t = _get_time_value_position([FIRST:STEP:LAST...], ts)
+		return arr.data[idxs1..., t, idxs2...]
+	end
+end
+
+function Base.getindex(arr::TimestepArray{VariableTimestep{TIMES}, T_data, N, ti}, idxs::Union{TimestepValue{T_time}, AnyIndex}...) where {T_data, N, ti, TIMES, T_time}
+	_single_index_check(arr.data, idxs)
+	idxs1, ts, idxs2 = split_indices(idxs, ti)
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		t = _get_time_value_position(TIMES, ts)
+		return arr.data[idxs1..., t, idxs2...]
+	end
+end
+
+function Base.getindex(arr::TimestepArray{FixedTimestep{FIRST, STEP}, T, N, ti}, idxs::Union{TimestepIndex, AnyIndex}...) where {T, N, ti, FIRST, STEP}
+	_single_index_check(arr.data, idxs)
+	idxs1, ts, idxs2 = split_indices(idxs, ti)
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		t = ts.index
+		_index_bounds_check(arr.data, ti, t)
+		return arr.data[idxs1..., t, idxs2...]
+	end
+end
+
+function Base.getindex(arr::TimestepArray{VariableTimestep{TIMES}, T, N, ti}, idxs::Union{TimestepIndex, AnyIndex}...) where {T, N, ti, TIMES}
+	_single_index_check(arr.data, idxs)
+	idxs1, ts, idxs2 = split_indices(idxs, ti)
+	if isa(ts, Colon)
+		return arr.data[idxs1..., :, idxs2...]
+	else
+		t = ts.index
+		_index_bounds_check(arr.data, ti, t)
+		return arr.data[idxs1..., t, idxs2...]
+	end
 end
 
 function Base.setindex!(arr::TimestepArray{FixedTimestep{FIRST, STEP}, T, N, ti}, val, idxs::Union{FixedTimestep{FIRST, STEP, LAST}, AnyIndex}...) where {T, N, ti, FIRST, STEP, LAST}
 	idxs1, ts, idxs2 = split_indices(idxs, ti)
-	setindex!(arr.data, val, idxs1..., ts.t, idxs2...)
+	if isa(ts, Colon)
+		setindex!(arr.data, val, idxs1..., :, idxs2...)
+	else
+		setindex!(arr.data, val, idxs1..., ts.t, idxs2...)
+	end
 end
 
 function Base.setindex!(arr::TimestepArray{VariableTimestep{TIMES}, T, N, ti}, val, idxs::Union{VariableTimestep{TIMES}, AnyIndex}...) where {T, N, ti, TIMES}
 	idxs1, ts, idxs2 = split_indices(idxs, ti)
-	setindex!(arr.data, val, idxs1..., ts.t, idxs2...)
+	if isa(ts, Colon)
+		setindex!(arr.data, val, idxs1..., :, idxs2...)
+	else
+		setindex!(arr.data, val, idxs1..., ts.t, idxs2...)
+	end
 end
```

Comment:
Also combine these two, if it works to also have the `TS` in the `idxs` (sorry I know that a lot of these pairs of methods existed before this PR, but I'm realizing if we can simplify we definitely should especially since we are adding so many new methods and it's hard to keep track of!)

Reply:
Making a new Issue and PR to remove duplicate functions